analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs

```diff
@@ -59,28 +58,76 @@ protected RequestsWithExcessiveLengthBase(System.Resources.ResourceManager rspec
         }
 
         protected override void Initialize(ParameterLoadingAnalysisContext context) =>
-            context.RegisterSyntaxNodeActionInNonGenerated(Language.GeneratedCodeRecognizer,
+            context.RegisterCompilationStartAction(
                 c =>
                 {
-                    if (!IsEnabled(c.Options))
-                    {
-                        return;
-                    }
-
-                    if (c.Node is TAttributeSyntax attribute
-                        && (attribute.IsKnownType(KnownType.Microsoft_AspNetCore_Mvc_DisableRequestSizeLimitAttribute, c.SemanticModel)
-                            || IsInvalidRequestSizeLimit(attribute, c.SemanticModel)
-                            || IsInvalidRequestFormLimits(attribute, c.SemanticModel)))
-                    {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, c.Node.GetLocation()));
-                    }
-                },
-                Language.SyntaxKind.Attribute);
+                    var attributesOverTheLimit = new ConcurrentDictionary<SyntaxNode, Attributes>();
+
+                    c.RegisterSyntaxNodeActionInNonGenerated(Language.GeneratedCodeRecognizer,
+                        cc =>
+                        {
+                            if (!IsEnabled(c.Options))
+                            {
+                                return;
+                            }
+
+                            var attribute = (TAttributeSyntax)cc.Node;
+
+                            if (attribute.IsKnownType(KnownType.Microsoft_AspNetCore_Mvc_DisableRequestSizeLimitAttribute, cc.SemanticModel))
+                            {
+                                cc.ReportDiagnosticWhenActive(Diagnostic.Create(rule, attribute.GetLocation()));
+                            }
+
+                            if ((IsInvalidRequestSizeLimit(attribute, cc.SemanticModel)
+                                 || IsInvalidRequestFormLimits(attribute, cc.SemanticModel))
+                                && GetMethodOrClassDeclaration(attribute, cc.SemanticModel) is { } delcaration)
+                            {
```

Comment:
this makes the overall method a bit complex, please move to a separate method, maybe named "CollectAttributesOverTheLimit"

the the pair method in EndAction could be extracted in "ReportOnCollectedAttributes"

Reply:
(also, this was hinted in the SonarCloud cognitive complexity issue)